### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by mqudsi**
Just some minor housekeeping to silence some build warnings that were bothering me; unless the warnings for the unused result of 'write' were intended so that they could be properly addressed (how?) at some point in the future.

### CODE REVIEW

File: src/path.cpp [hunk 1]
```diff
@@ -275,7 +275,7 @@ static void maybe_issue_path_warning(const wcstring &which_dir, const wcstring &
         debug(0, _(L"The error was '%s'."), strerror(saved_errno));
         debug(0, _(L"Please set $%ls to a directory where you have write access."), env_var);
     }
-    write(STDERR_FILENO, "\n", 1);
+    write_ignore(STDERR_FILENO, "\n", 1);
```
Comment (krader1961) [inline src/path.cpp]: The previous change to explicitly initialize some size_t vars is fine. I'm less happy with this change. Prior to this there are only four locations that use `write_ignore()`. I'd like input from others but for me I would prefer to modify the callers to explicitly discard the return value. That is, replace `write_ignore(...)` with `(void)write(...)`. While the name of the former function is reasonably clear it adds an unneeded level of indirection and can thus cause confusion.

Comment (mqudsi) [inline src/path.cpp]: I agree with your assessment; I did not feel it was my place as such a newcomer to suggest silencing the warnings entirely and just wanted to unify the warning suppression to use the existing `read|write_ignore` method. If we have no intention of handling write to stderr failure somehow, then it makes no sense leaving the `write_ignore` calls lying around.

Comment (krader1961) [inline src/path.cpp]: In practice there isn't anything fish can do that would be sensible if a write to stdout or stderr fails. The most common source of an error is the tty being closed, a full filesystem (errno ENOSPC) or the read side of a pipe being closed (errno EPIPE). Often the simplest way to deal with these failures is to ignore them and simply let the script run to completion. The alternative is to exit early when the first error occurs.

**Comment by krader1961** [comment]
Merged although I replaced the _ignore variants with there wrapped function prefaced by `(void)`.

**Pull request closed without merging**